src/Core/Services/IOrganizationService.cs

```diff
@@ -39,6 +39,16 @@ public interface IOrganizationService
         OrganizationUserType type, bool accessAll, string externalId, IEnumerable<CollectionAccessSelection> collections, IEnumerable<Guid> groups);
     Task<IEnumerable<Tuple<OrganizationUser, string>>> ResendInvitesAsync(Guid organizationId, Guid? invitingUserId, IEnumerable<Guid> organizationUsersId);
     Task ResendInviteAsync(Guid organizationId, Guid? invitingUserId, Guid organizationUserId, bool initOrganization = false);
+    /// <summary>
+    /// Accepts a user into an organization based on the provided invitation token and user information.
+    /// Validates the invitation token, checks for existing memberships, and ensures email correspondence before acceptance.
+    /// Marks the user's email as verified upon successful acceptance.
+    /// </summary>
+    /// <param name="organizationUserId">The unique identifier of the organization user.</param>
+    /// <param name="user">Information about the user accepting the invitation.</param>
+    /// <param name="token">Token associated with the invitation.</param>
+    /// <param name="userService">Service for user-related operations.</param>
+    /// <returns>The accepted OrganizationUser.</returns>
```

Comment:
This may be a nit, or just a style difference, but I think this is too much info, and unnecessary explanation on the parameters. I suggest shorter & simpler, e.g.

```suggestion
    /// <summary>
    /// Moves an OrganizationUser into the Accepted status and marks their email as verified.
    /// This method is used where the user has clicked the invitation link sent by email.
    /// </summary>
    /// <param name="token">The token embedded in the email invitation link</param>
    /// <returns>The accepted OrganizationUser.</returns>
```

One thing I don't know is if we need to describe all the other params as well. They still show up in the method signature so I'm inclined to only comment on those that are unclear, but feel free to disagree on that.

Reply:
I tested it and (at least on Rider) all the parameters showed up but only `token` had a description.